sdk/keyvault/azure-security-keyvault-keys/src/private/keyvault_protocol.hpp

```diff
@@ -21,6 +21,10 @@
 #include <string>
 #include <vector>
 
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+  class KeyClient;
+}}}} // namespace Azure::Security::KeyVault::Keys
```

Comment:
@antkmsft maybe you have better insights on how to manage this cyclical dependency, similar to https://github.com/Azure/azure-sdk-for-cpp/pull/2625#discussion_r669405309

Reply:
Discussed with @CaseyCarter offline, both patterns are OK.